common/lib/xmodule/xmodule/modulestore/split_mongo/split.py

```diff
@@ -795,6 +800,22 @@ def _add_cache(self, course_version_guid, system):
         """
         if self.request_cache is not None:
             self.request_cache.data.setdefault('course_cache', {})[course_version_guid] = system
+            number_courses_in_cache = len(self.request_cache.data['course_cache'])
+            if number_courses_in_cache > self.max_num_courses_in_cache:
+                # We shouldn't have any scenarios where there are many
+                # courses in the request cache. If there are, it's probably
+                # indicative of a leak. In this case, we should log that here
+                # with a traceback.
+                log.warning((
+                    "There are more than {max_num_courses} ({num_courses}) "
+                    "courses in the request cache. This may be indicative "
+                    "of a memory leak. Traceback:\n"
+                    "{traceback}"
+                ).format(
+                    max_num_courses=self.max_num_courses_in_cache,
+                    num_courses=number_courses_in_cache,
+                    traceback="".join(traceback.format_stack()[-10:]),
+                ))
```

Comment:
If you're seeing this situation, it might also be helpful to log which courses are here (up to some capped number).